internal/sync/cache.go

```diff
@@ -0,0 +1,362 @@
+package sync
+
+import (
+	"errors"
+	"fmt"
+	"github.com/argoproj/gitops-engine/pkg/diff"
+	"github.com/cespare/xxhash/v2"
+	"github.com/numaproj-labs/numaplane/internal/kubernates"
+	"github.com/numaproj-labs/numaplane/internal/shared"
+	"go.uber.org/zap"
+	"net"
+	"net/url"
+	"os/exec"
+	"strconv"
+	"strings"
+	"sync"
+	"syscall"
+	"time"
+
+	clustercache "github.com/argoproj/gitops-engine/pkg/cache"
+	"github.com/argoproj/gitops-engine/pkg/health"
+	"github.com/argoproj/gitops-engine/pkg/utils/kube"
+	"github.com/numaproj-labs/numaplane/api/v1alpha1"
+	"golang.org/x/sync/semaphore"
+	v1 "k8s.io/api/core/v1"
+	kerrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/rest"
+)
+
```

Comment:
Can we have a high level description somewhere to indicate what this cache is all about?